Wikipedia/Code/ArticleSurveyTimerController.swift

```diff
@@ -0,0 +1,125 @@
+import Foundation
+
+/// Manages the timer used to display survey announcements on articles
+final class ArticleSurveyTimerController {
+
+    // MARK: - Public Properties
+
+    var timerFireBlock: ((SurveyAnnouncementsController.SurveyAnnouncementResult) -> Void)?
+
+    // MARK: - Private Properties
+
+    private let articleURL: URL
+    private let surveyController: SurveyAnnouncementsController
```

Comment:
Do we need this? We pass in `SurveyAnnoucementsController.shared` into the initializer - doesn't that mean we could just call `SurveyAnnoucementsController.shared` in this class instead?

Reply:
yeah, I have a bad habit of setting things up for easier unit testing but not following through on making the tests themselves. But with this method we could inject our own flavor of SurveyAnnouncementsController that hard codes announcements. But I still don't expect we'd be able to do this without further decoupling, i.e. having this be a protocol that `SurveyAnnouncementsController` conforms to or marking `SurveyAnnouncementsController` as not-final so we can subclass with our own announcements. I'm fine with taking out dependency injection until we actually use it.

Reply:
Thanks for pointing this out - I just noticed I forgot to change: https://github.com/wikimedia/wikipedia-ios/blob/aced47d626cb3765c7e587168ae5ac7b66e5912d/Wikipedia/Code/ArticleSurveyTimerController.swift#L28 to use `surveyController`, so I'll do that now.

@tonisevener I think maintaining the classical dependency injection here, even if we're unable to immediately follow through on tests, more clearly establishes the responsibilities  and expectations when using this (or any) controller. Overusing singletons definitely contributes to unclear code paths and potentially confusing global mutable state.

Unless either of you feel very strongly that we should remove it, I think it's stronger in the long run to leave the dependency injection here as-is.

Reply:
I was just coming back to point out the `.shared` still in there.

Reply:
Once that's updated, I'm good to merge.

Reply:
👍 just removed that singleton usage and also removed some extraneous indentation. Feel free to merge when you're comfortable with it!